python/dgl/convert.py

```diff
@@ -17,397 +18,211 @@
+
+    Create a graph on the first GPU card with data type int32.
 
-    if isinstance(data, DGLHeteroGraph):
-        return data.astype(idtype).to(device)
+    >>> g = dgl.graph((src_ids, dst_ids), idtype=torch.int32, device='cuda:0')
 
-    if card is not None:
-        dgl_warning("Argument 'card' will be deprecated. "
-                    "Please use num_nodes={} instead.".format(card))
-        num_nodes = card
+    See Also
+    --------
+    from_scipy
+    from_networkx
+    """
+    # Sanity check
+    utils.check_type(data, tuple, 'data', skip_none=False)
+    if len(data) != 2:
+        raise DGLError('Expect data to have length 2, got {:d}'.format(len(data)))
+
+    src_type = type(data[0])
+    dst_type = type(data[1])
+    if src_type != dst_type:
+        raise DGLError('Expect the source and destination node IDs to have the same type, ' \
+                       'got {} and {}'.format(src_type, dst_type))
+
+    if len(data[0]) != len(data[1]):
+        raise DGLError('Expect the source and destination node IDs to have the same length, ' \
+                       'got {:d} and {:d}'.format(len(data[0]), len(data[1])))
+
+    if F.is_tensor(data[0]):
+        src_dtype = F.dtype(data[0])
+        dst_dtype = F.dtype(data[1])
+        if src_dtype != dst_dtype:
+            raise DGLError('Expect the source and destination node tensors to have the same ' \
+                           'data type, got {} and {}'.format(src_dtype, dst_dtype))
+        if src_dtype not in [F.int32, F.int64]:
+            raise DGLError('Expect the node-tensors to have data type int32 or int64, '
+                           'got {}'.format(src_dtype))
+        src_ctx = F.context(data[0])
+        dst_ctx = F.context(data[1])
+        if src_ctx != dst_ctx:
+            raise DGLError('Expect the source and destination node tensors to have the same ' \
+                           'context, got {} and {}'.format(src_ctx, dst_ctx))
+    elif isinstance(data[0], Iterable):
+        # NaN/Inf values cannot appear in int32/int64 tensors
+        utils.detect_nan_in_iterable(data[0], 'the source node IDs')
+        utils.detect_inf_in_iterable(data[0], 'the source node IDs')
+        utils.detect_nan_in_iterable(data[1], 'the destination node IDs')
+        utils.detect_inf_in_iterable(data[1], 'the destination node IDs')
+    else:
+        raise DGLError('Expect sequences (e.g., list, numpy.ndarray) or tensors for data, ' \
+                       'got {}'.format(type(data[0])))
+
+    if ntype is not None:
+        raise DGLError('The ntype argument is deprecated for dgl.graph. To construct ' \
+                       'a graph with named node types, use dgl.heterograph.')
+    if etype is not None:
+        raise DGLError('The etype argument is deprecated for dgl.graph. To construct ' \
+                       'a graph with named edge types, use dgl.heterograph.')
+
+    if num_nodes is not None and not (isinstance(num_nodes, int) and num_nodes >= 0):
```

Comment:
What if the user want to create an empty graph by passing num_nodes = 0.

Reply:
That can be realized with `dgl.graph(([], []), num_nodes=0)`